hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/AccountKeyLookup.java

```diff
@@ -46,4 +47,28 @@ public interface AccountKeyLookup {
      * @return key if successfully fetched or failureReason for failure
      */
     KeyOrLookupFailureReason getKeyIfReceiverSigRequired(final AccountID idOrAlias);
+
+    /**
+     * Fetches the contract's key from given contractID. If the key could not be fetched as the
+     * given contractID is invalid or doesn't exist provides information about the failure
+     * failureReason. If there is no failure failureReason will be null.
+     *
+     * @param idOrAlias contract id whose key should be fetched
+     * @return key if successfully fetched or failureReason for failure
+     */
+    KeyOrLookupFailureReason getKey(final ContractID idOrAlias);
```

Comment:
Null annotations missing for return param and method params (Nullable / NonNull)

Reply:
Done